src/app/complex/complex-results/complex-results.component.ts

```diff
@@ -194,4 +242,47 @@ export class ComplexResultsComponent implements OnInit, AfterViewInit {
   set interactorTypeFilter(value: string[]) {
     this._interactorTypeFilter = value;
   }
+
+  public get allInteractorsInComplexSearch(): Interactor[] {
+    return this._allInteractorsInComplexSearch;
+  }
+
+  setListView() {
+    this.DisplayType = this.LIST_VIEW;
+    this.toast = this.notificationService.complexNavigatorAnnouncement();
+    this.reloadPage();
+  }
+
+  setComplexNavigatorView() {
+    this.DisplayType = this.COMPLEX_NAVIGATOR_VIEW;
+    this.reloadPage();
+    if (!!this.toast) {
+      this.notificationService.closeAnnouncement(this.toast.toastId);
+      this.toast = null;
+    }
+  }
+
+
+  setFirstDisplayType(): void {
+    if (this._complexSearch.elements.length === 1) {
+      const complexId = this._complexSearch.elements[0].complexAC;
+      if (!!complexId) {
+        // For some reason this is needed so the navigate call works
+        this.router.routeReuseStrategy.shouldReuseRoute = function () {
+          return false;
+        };
+        this.router.navigate(['/complex', complexId]);
+      }
+    } else if (this._complexSearch.elements.length < this._navigatorPageSize) {
```

Comment:
A few things here.

1. `this._complexSearch.elements` are the elements in the page, so they will always be 15 or 20. We should do 
```
if (this._complexSearch.totalNumberOfResults <= this._navigatorPageSize) {
```

2. As you can see in the code above, we should use `<=` instead of `<`. If there are 20 results or less, we want the navigator view.

3. More importantly, the way this is set up, we always check the length of the elements after a search. That means that if someone goes to `http://localhost:4200/complex/search?query=KMN&page=1#view_list`, which has the fragment `#view_list`, because there are 9 results, they get redirected to the navigator view. We should only do this if the fragment is not set, which happens for urls like `http://localhost:4200/complex/search?query=KMN&page=1` or when people initially search. To fix this we need to move this if/else inside the if below:
```
    if (!this.DisplayType) {
      if (this._complexSearch.totalNumberOfResults <= this._navigatorPageSize) {
        this.setComplexNavigatorView();
      } else {
        this.setListView();
      }
    }
```